design/ceph-config-updates.md

```diff
@@ -0,0 +1,258 @@
+  # temporary overrides will not be possible.
+  config:
+    # Each key in the 'config' section represents a config file section. 'global' is likely to be
+    # the only section which is modified; however, daemons can have their config overridden
+    # explicitly if desired.
+    # global will add/override config for all Ceph daemons
+    global:
+      # The below "osd_pool_default..." settings make the default pools created have no replication
+      # and should be removed for production clusters, as this could impact data fault tolerance.
+      osd_pool_default_size: 1
+      osd_pool_default_min_size: 1
+    # mon will add/override config for all mons
+    mon:
+      mon_cluster_log_file: "/dev/stderr"
+    # osd.0 will add/override config only for the osd with ID 0 (zero)
+    osd.0:
+      debug_osd: 10
+   # ...
+```
+**Note on the above:** all values under config are reported to Ceph as strings, but the yaml should
+support integer values as well if at all possible
+
+As stated in the example yaml, above, the 'config' section adds or overrides values in the Ceph
+config whenever the Ceph operator starts and whenever the user updates the cluster CRD. Ceph
+Luminous does not have a centralized config, so the overrides from this section will have to be set
+on the command line. For Ceph Mimic and above, the mons have a centralized config which will be used
+to set/override configs. Therefore, for Mimic+ clusters, the user may temporarily override values
+set here, and those values will be reset to the `spec:config` values whenever the Ceph operator is
+restarted or the cluster CRD is updated.
+
+#### Additional configs for test/play environments
+
+Test (especially integration tests) may need to specify `osd pool default size = 1` and
+`osd pool default min size = 1` to support running clusters with only one osd. Test environments
+would have a means of doing this fairly easily using the config override capability. These values
+should not be set to these low values for production clusters, as they may allow admins to create
+their own pools which are not fault tolerant accidentally.
+
+There is an option to set these values automatically for clusters which run with only one osd or to
+set this value for clusters with a number of osds less than the default programmatically within
+Rook's operator; however, this adds an additional amount of code flow complexity which is
+unnecessary except in the integration test environments or in minimal demo environments. A middle
+ground proposed herein is to add `osd pool default {,min} size = 1` overrides to the example cluster
+CRD so that users "just trying out Rook" still get today's easy experience but where they can be
+easily removed for production clusters that should not run with potentially dangerous settings.
+
+### Changes to Ceph mons
+
+The current method of starting mons where `mon-a` has an initial member of `a`, `mon-b` initial
```

Comment:
Changing the mon startup method would be good to do independently from the settings changes. @leseb was also discussing this approach today while on another topic.